internal/gazelle.bash.in

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# Copyright 2017 The Bazel Authors. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#    http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+@@GENERATED_MESSAGE@@
+
+set -euo pipefail
+
+RUNNER_LABEL=@@RUNNER_LABEL@@
+GAZELLE_SHORT_PATH=@@GAZELLE_SHORT_PATH@@
+GAZELLE_LABEL=@@GAZELLE_LABEL@@
+ARGS=@@ARGS@@
+
+# is_bazel_run returns true (0) if this script was invoked with "bazel run".
+function is_bazel_run {
+  # This variable is set by the bazel wrapper script.
+  [ -n "${BAZEL_REAL-}" ]
+}
+
+# find_repo_root prints the absolute path to the repository root. This is
+# discovered by locating the WORKSPACE file in a parent directory, and
+# deferencing symlinks. If no WORKSPACE file is found, this function prints
+# an error and exits.
+function find_repo_root {
+  pushd . &>/dev/null
+  while [ $(pwd) != / -a ! -f WORKSPACE ]; do
```

Comment:
Would be safer to test if cd .. fails to change pwd, not all file systems end at /